### PR TITLE
Replace LayerNorm with a custom layer in ALBERT to avoid inf loss

### PR DESCRIPTION
**Pull request by agobeljicTT**
### Issue
Workaround for https://github.com/tenstorrent/tt-metal/issues/41127

### Bug Description
There is a regression in `batch_norm_training` which affects LayerNorm.

### What's Changed
This PR implements a custom TTLayerNorm that should bypass the call to the `batch_norm_training` and force manual ops which (when executed on their own) do not have correctness issues.

### Environment
- **Commit/Version:** N/A
- **Hardware:** N300

### Checklist
- [ ] Bug is reproducible before the fix
- [ ] Fix resolves the bug
- [ ] Tests pass and updated if needed
- [ ] No new bugs introduced
- [ ] Documentation updated if needed

### CODE REVIEW

File: blacksmith/experiments/torch/albert/test_albert_finetuning.py [hunk 1]
```diff
@@ -18,6 +18,38 @@
 from blacksmith.tools.device_manager import DeviceManager
 from blacksmith.tools.logging_manager import TrainingLogger
 from blacksmith.tools.reproducibility_manager import ReproducibilityManager
+from blacksmith.tools.workaround_utils import TTLayerNorm
+
+
+def replace_layernorm(module):
```
Comment (pglusacTT) [inline blacksmith/experiments/torch/albert/test_albert_finetuning.py]: This should be placed inside workaround utils.

**Review by pglusacTT — APPROVED** [review]
Left only one comment, other than that this is great!
Thanks a lot for doing this, amazing job!

File: blacksmith/experiments/torch/albert/test_albert_finetuning.py [hunk 2]
```diff
@@ -77,6 +109,9 @@ def train(
     if config.resume_from_checkpoint:
         checkpoint_manager.load_checkpoint()
 
+    # TODO(agobeljic): https://github.com/tenstorrent/tt-metal/issues/41127
+    replace_layernorm(model)
```
Comment (mmilosevicTT) [inline blacksmith/experiments/torch/albert/test_albert_finetuning.py:82]: Do we need to explicitly unfreeze parameters?

File: blacksmith/tools/workaround_utils.py [hunk 3]
```diff
@@ -37,3 +38,40 @@ def transform_labels(labels, ignored_index, vocab_size):
     expected_output = F.one_hot(labels, num_classes=vocab_size)
 
     return expected_output, labels_mask
+
+
+# Custom LayerNorm for TT-Forge. Necessary because of Albert experiment getting inf loss.
+class TTLayerNorm(nn.Module):
+    def __init__(self, normalized_shape, eps=1e-5, elementwise_affine=True, device=None, dtype=None):
+        super().__init__()
+        # Handle cases where normalized_shape is an int
```
Comment (mmilosevicTT) [inline blacksmith/tools/workaround_utils.py]: nit: Use full sentences
```suggestion
        # Handle cases where `normalized_shape` is an `int`.
```

Comment (mmilosevicTT) [reply blacksmith/tools/workaround_utils.py]: Ditto for all other comments.

File: blacksmith/tools/workaround_utils.py [hunk 4]
```diff
@@ -37,3 +38,40 @@ def transform_labels(labels, ignored_index, vocab_size):
     expected_output = F.one_hot(labels, num_classes=vocab_size)
 
     return expected_output, labels_mask
+
+
+# Custom LayerNorm for TT-Forge. Necessary because of Albert experiment getting inf loss.
```
Comment (mmilosevicTT) [inline blacksmith/tools/workaround_utils.py]: I would put link to the issue without this comment here.

**Review by mmilosevicTT — APPROVED** [review]
LGTM

File: blacksmith/experiments/torch/albert/test_albert_finetuning.py (hunk 2 above)
Comment (agobeljicTT) [reply blacksmith/experiments/torch/albert/test_albert_finetuning.py:82]: we need to freeze them actually

Comment (agobeljicTT) [reply blacksmith/experiments/torch/albert/test_albert_finetuning.py:82]: good catch

File: blacksmith/tools/workaround_utils.py (hunk 4 above)
Comment (agobeljicTT) [reply blacksmith/tools/workaround_utils.py]: the thing is we don't have a proper issue yet, we haven't found a minimal repro, it isn't trivial.
i can remove the comment nevertheless.

**Review by pglusacTT — APPROVED** [review]
